src/formats.rs

```diff
@@ -44,6 +40,19 @@ impl PkgFmt {
             PkgFmt::Zip => PkgFmtDecomposed::Zip,
         }
     }
+
+    /// List of possible file extensions for the format.
+    pub fn extensions(self) -> &'static [&'static str] {
+        match self {
+            PkgFmt::Tar => &["tar"],
+            PkgFmt::Tbz2 => &["tbz2", "tar.bz2"],
+            PkgFmt::Tgz => &["tgz", "tar.gz"],
+            PkgFmt::Txz => &["txz", "tar.xz"],
+            PkgFmt::Tzstd => &["tzstd", "tzst", "tar.zst"],
+            PkgFmt::Bin => &["exe"],
```

Comment:
I think it should also have a `""` here for elf, though I could be wrong:

```suggestion
            PkgFmt::Bin => &["exe", ""],
```

Reply:
Yeah, we can't really do that because the URL template will have

```
https://blabla/file.{ archive-format }
```

so if the extension is empty it works out to


```
https://blabla/file.
```

which isn't what's wanted.

It's a bit awkward but we can't really change the format now, and so that syntax wouldn't be in use anyway for plain elf bins.

Reply:
So for plain elf bins on unix, we will just skip the postfix?

Reply:
`{archive-format}` is just for the URL, so for plain elf bins you shouldn't have it at all in the URL template. Having it in there doesn't work now, and it still won't work with this.

Reply:
Got it.